lnd_methods/offchain/estimate_route_fee.js

```diff
@@ -0,0 +1,91 @@
+const asyncAuto = require('async/auto');
+const {returnResult} = require('asyncjs-util');
+const {isStringLiteral, isNumericLiteral} = require('typescript');
+
+/** Estimate routing fees based on an invoice.
+
+  Requires `offchain:read` permission
+
+  {
+    lnd: <Authenticated LND API Object>
+    request: <BOLT 11 Payment Request String>
+    timeout: <Route Fee Estimate Timeout in Milliseconds Number> 
+  }
+
+  @returns via cbk or Promise
+  {
+    routing_fee_msat: <Route Fee Millitokens String>,
+    time_lock_delay: <Time Lock Block Height Delay String>
+    failure_reason: <Failure Reason String>
```

Comment:
if it errors when it fails we can omit this

Reply:
what does this mean?

Reply:
we aren't returning a failure as a success here, failure is an error to the cbk, so there's no failure reason

Reply:
I see, sounds good, thanks